pkg/name/name_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package name
+
+import (
+	"reflect"
+	"testing"
+
+	"istio.io/operator/pkg/util"
+)
+
+func TestGetFromTreePath(t *testing.T) {
+	type args struct {
+		inputTree map[string]interface{}
+		path      util.Path
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    interface{}
+		found   bool
+		wantErr bool
+	}{
+		{
+			name: "found string node",
+			args: args{
+				inputTree: map[string]interface{}{
+					"k1": "v1",
```

Comment:
These test inputs are a bit small - it would be better to have a greater depth and more keys. One way to do this easily is to make inputTree a yaml string (then you can easily do maps and lists) and yaml.unmarshal it when you run the test.